prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -1352,6 +1355,106 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
+        txn: DataAccessTransaction? = null
+    ): List<Permission> {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        return ctx.select(PERMISSION.asterisk())
+            .from(PERMISSION)
+            .join(PERMISSION_ORGANIZATION)
+            .on(PERMISSION.PERMISSION_ID.eq(PERMISSION_ORGANIZATION.PERMISSION_ID))
+            .where(
+                PERMISSION_ORGANIZATION.ORGANIZATION_NAME.eq(organizationName)
+                    .and(PERMISSION.ENABLED.eq(true))
+            )
+            .limit(100)
+            .fetchInto(Permission::class.java)
+    }
+
+    /**
+     * Returns a permission
+     * @param id an exact permission id
+     * @param txn an optional database transaction
+     */
+    fun fetchPermissionById(
+        id: Int,
+        txn: DataAccessTransaction? = null
+    ): Permission? {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        return ctx.selectFrom(PERMISSION)
+            .where(
+                PERMISSION.PERMISSION_ID.eq(id)
+            )
+            .fetchOne()
+            ?.into(Permission::class.java)
+    }
+
+    /**
+     * Updates a permission
+     * @param id an exact permission id
+     * @param permission type of Permission
+     * @param txn an optional database transaction
+     */
+    fun updatePermissionById(
+        id: Int,
+        permission: Permission,
+        txn: DataAccessTransaction? = null
+    ) {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        ctx.update(PERMISSION)
+            .set(PERMISSION.ENABLED, permission.enabled)
+            .where(PERMISSION.PERMISSION_ID.eq(id))
+            .execute()
+    }
+
+    /**
+     * Returns a permission
+     * @param userName the user name of the creator
+     * @param permission type of Permission
+     * @param txn an optional database transaction
+     */
+    fun insertPermission(
+        user: String,
+        permission: Permission,
+        txn: DataAccessTransaction? = null
+    ): Int {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        return ctx.insertInto(PERMISSION)
+            .set(
+                PERMISSION.PERMISSION_ID,
+                DSL.defaultValue(PERMISSION.PERMISSION_ID)
+            )
+            .set(PERMISSION.NAME, permission.name)
+            .set(PERMISSION.DESCRIPTION, permission.description)
+            .set(PERMISSION.CREATED_BY, user)
+            .set(PERMISSION.CREATED_AT, OffsetDateTime.now())
+            .returningResult(PERMISSION.PERMISSION_ID)
+            .fetchOne()
+            ?.value1()
+            ?: error("Fetch error")
+    }
+
+    /**
+     * Deletes a permission
+     * @param id an exact permission id
+     * @param txn an optional database transaction
+     */
+    fun deletePermissionById(
+        id: Int,
+        txn: DataAccessTransaction? = null
+    ) {
+        val ctx = if (txn != null) DSL.using(txn) else create
+        ctx.deleteFrom(PERMISSION)
```

Comment:
@ahay-agile6 Do you know if we have a soft-delete vs hard-delete rule/norm? Always something I consider when seeing delete functions

Reply:
@kevinhaube I think were good since we have an enable flag for each permission. To address your comment about the norm I noticed that we mostly have hard deletes.

Reply:
Awesome! Thanks for checking on this